app/models/SupplementaryRequest.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import cats.data.{EitherNec, NonEmptyChain}
+import cats.implicits.catsSyntaxTuple2Parallel
+import pages.{HasSupplementaryUnitUpdatePage, SupplementaryUnitUpdatePage}
+import play.api.libs.json.{Json, OFormat}
+import queries.MeasurementQuery
+
+final case class SupplementaryRequest(
+  eori: String,
+  recordId: String,
+  hasSupplementaryUnit: Option[Boolean] = None,
+  supplementaryUnit: Option[String] = None,
+  measurementUnit: Option[String] = None
+)
+
+object SupplementaryRequest {
+
+  implicit lazy val format: OFormat[SupplementaryRequest] = Json.format
+
+  def build(answers: UserAnswers, eori: String, recordId: String): EitherNec[ValidationError, SupplementaryRequest] = {
+    val hasSupplementaryUnitEither = getHasSupplementaryUnit(answers, recordId)
+    val supplementaryUnitEither    = getSupplementaryUnit(answers, recordId)
+    val measurementUnitEither      = getMeasurementUnit(answers, recordId)
+
+    hasSupplementaryUnitEither
+      .flatMap { hasSupplementaryUnit =>
+        if (hasSupplementaryUnit) {
+          (supplementaryUnitEither, measurementUnitEither).parMapN { (supplementaryUnit, measurementUnit) =>
+            SupplementaryRequest(
+              eori = eori,
+              recordId = recordId,
+              hasSupplementaryUnit = Some(true),
+              supplementaryUnit = Some(supplementaryUnit),
+              measurementUnit = Some(measurementUnit)
+            )
+          }
+        } else {
+          Right(
+            SupplementaryRequest(
+              eori = eori,
+              recordId = recordId,
+              hasSupplementaryUnit = Some(false),
+              supplementaryUnit = None,
+              measurementUnit = None
+            )
+          )
+        }
+      }
+      .orElse {
+        (supplementaryUnitEither, measurementUnitEither).parMapN { (supplementaryUnit, measurementUnit) =>
+          SupplementaryRequest(
```

Comment:
Does it make sense to do anything in this case? If they haven't answered HasSupplementaryUnit we should probably just be returning the error that is in the Left already? 

(so not have the .orElse case at all?)

Reply:
It should have .OrElse case, as we have a journey without HasSupplementaryUnit (ie., Just supplementary unit) as well.